src/libstore/unix/build/local-derivation-goal.cc

```diff
@@ -1171,6 +1173,12 @@ void LocalDerivationGoal::initEnv()
 
     /* Trigger colored output in various tools. */
     env["TERM"] = "xterm-256color";
+
+    /* Set up authentication tunneling for builtin:fetchurl. FIXME:
+       maybe we want to support this for arbitrary fixed-output
+       derivations. */
```

Comment:
```suggestion
    /* Set up authentication tunneling for builtin:fetchurl.
       We generally do not want to pass credentials to arbitrary
       fixed output derivations, because that makes it possible
       for any dependency (however deep down) to steal them. */
```

We'd need to come up with a scheme to safely pass authorization (with letter _z_) to specific derivations in such a way that the user controls where their credentials go; perhaps through some sort of unforgeable token that's passed around explicitly in expressions? Something for another day.

Reply:
OT: notably we can't store the unforgeable token in the drv itself, because that would expose it to be read with `readFile foo.drvPath`.

This seems to be another instance where we should be tracking transient derivation metadata, just like we should have been doing with `meta.timeout` or perhaps the licenses.